src/app/src/main/kotlin/org/imperial/mrc/hint/Extensions.kt

```diff
@@ -51,16 +53,22 @@ fun Response.asResponseEntity(): ResponseEntity<String> {
     return try {
         val body = this.body().asString("application/json")
         val json = ObjectMapper().readTree(body)
-        if (!json.has("status")) {
+        if (!json.has("status") && !json.has("result")) {
             throw IOException()
         }
-
-        ResponseEntity.status(httpStatus)
-                .contentType(MediaType.APPLICATION_JSON)
-                .body(body)
+        return if (json.has("status")) {
+            // this is a hintr response, so already conforms to our response schema
+            ResponseEntity.status(httpStatus)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .body(body)
+        } else {
+            // this is an ADR response, so convert to our response schema
+            SuccessResponse(json["result"])
```

Comment:
Is this going to behave correctly if ADR gives back something other than a 200?

Reply:
Er, no it won't! Thanks for spotting. Actually the ckan API always returns a 200, even if the request errors, but we should be looking at the `success` property to work out whether it succeeded or not: https://github.com/mrc-ide/hint/pull/273/commits/bd51356bca226ea6d422f0f94f9c527219aaf53c